PDA can be ctrl-clicked to remove the item in its pen slot

Irafas opened the pull request:
:cl: 
add: Players can now ctrl-click the PDA to remove the item in its pen slot
/:cl:

This just makes taking the pen, or other items, out of the PDA easier. Now you can do it with just one click and one hand.

Irafas committed:
Makes pens ez to remove from pda

Players can now ctrl click on the PDA to remove the pen without having to remove the ID first.

Irafas committed:
added instructions when examined.

Like how it tells the user they can use alt-click to remove contents, it now says they can use ctrl-click to remove items from the pen slot.

TBartl commented:
What happens if you try to pull a PDA by CTRL+Clicking it?

ExcessiveUseOfCobblestone commented:
Just switch the order in alt click.

Irafas commented:
@TBartl 
SHIT
I'm an idiot as I somehow forgot about that. Might as well close this PR in shame

@ExcessiveUseOfCobblestone 
Most people care more about removing the ID first so I think that's not going to be very useful. Also I found that there's a verb for removing the pen from a PDA already, so this PR isn't really necessary.

ExcessiveUseOfCobblestone commented:
At the end of the day you're just alt clicking twice either way lol

fludd12 commented:
Wait nooooo reopen this and make it work only when loc is not a turf

It's not like someone will pull it in their ID slot

Irafas committed:
PDA can now be dragged without dropping the pen.

When ctrl-clicked it now checks to see if the PDA is on a turf and will only output the pen if its not.

Irafas commented:
Alright should work now, Thanks @fludd12 for the solution.

ExcessiveUseOfCobblestone commented on code/game/objects/items/devices/PDA/PDA.dm:
```diff
@@ -93,7 +93,7 @@ GLOBAL_LIST_EMPTY(PDAs)
 		return
 	else
-		to_chat(user, "<span class='notice'>Alt-click to remove contents.</span>")
+		to_chat(user, "<span class='notice'>Alt-click to remove contents.<br>Ctrl-click to remove item from pen slot</span>")
```
Only if it's on you.

Cyberboss replied:
This line needs better wording

Irafas committed:
quick change to examine text.

Cyberboss commented on code/game/objects/items/devices/PDA/PDA.dm:
```diff
@@ -93,7 +93,7 @@ GLOBAL_LIST_EMPTY(PDAs)
 		return
 	else
-		to_chat(user, "<span class='notice'>Alt-click to remove contents.<br>Ctrl-click to remove item from pen slot</span>")
+		to_chat(user, "<span class='notice'>Alt-click to remove contents.<br>Ctrl-click, while its in an inventory, to remove the pen</span>")
```
Remove the comma after Ctrl-click

Irafas committed:
comma

ExcessiveUseOfCobblestone commented:
Feel free to use

```javascript
if(id)
    to_chat(user, "<span class='notice'>Alt-click to remove [id].</span>")
if(inserted_item && (!isturf(loc)))
    to_chat(user, "<span class='notice'>Ctrl-click to remove [inserted_item].</span>"
```

The secret "Alt-click to remove the pen if the pen isn't there" can be just that.

Irafas committed:
Improved the examine

ExcessiveUseOfCobblestone commented on code/game/objects/items/devices/PDA/PDA.dm:
```diff
@@ -92,8 +92,12 @@ GLOBAL_LIST_EMPTY(PDAs)
-		to_chat(user, "<span class='notice'>Alt-click to remove contents.</span>")
+
+	if(id)
+		to_chat(user, "<span class='notice'>Alt-click to remove the id.</span>")
```
Probably a wise decision.
